### PR TITLE
[Merged by Bors] - feat: support irreducible_def in to_additive

### PR DESCRIPTION
**Pull request by gebner**
---

This resolves all of the `irreducible_def` todos I could find.

- [x] depends on: #3395 [optional extra text]

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#3395
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Tactic/Eqns.lean [hunk 1]
```diff
@@ -37,7 +37,7 @@ initialize eqnsAttribute : NameMapExtension (Array Name) ←
     descr := "Overrides the equation lemmas for a declation to the provided list"
     add   :=  fun
     | _, `(attr| eqns $[$names]*) =>
-      pure <| names.map (fun n => n.getId)
+      names.mapM resolveGlobalConstNoOverload
```
Comment (digama0) [inline Mathlib/Tactic/Eqns.lean:40]:
```suggestion
      names.mapM resolveGlobalConstNoOverloadWithInfo
```

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#3395~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by fpvandoorn** [comment]
LGTM
Thanks!
bors merge

**Pull request closed without merging**